backend/src/main/java/com/htecgroup/skynest/service/impl/AuthServiceImpl.java

```diff
@@ -0,0 +1,92 @@
+package com.htecgroup.skynest.service.impl;
+
+import com.htecgroup.skynest.exception.UserException;
+import com.htecgroup.skynest.exception.UserExceptionType;
+import com.htecgroup.skynest.model.dto.UserDto;
+import com.htecgroup.skynest.model.email.Email;
+import com.htecgroup.skynest.model.entity.UserEntity;
+import com.htecgroup.skynest.model.request.UserPasswordResetRequest;
+import com.htecgroup.skynest.repository.UserRepository;
+import com.htecgroup.skynest.service.AuthService;
+import com.htecgroup.skynest.service.EmailService;
+import com.htecgroup.skynest.service.UserService;
+import com.htecgroup.skynest.util.EmailUtil;
+import com.htecgroup.skynest.util.JwtUtils;
+import lombok.AllArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class AuthServiceImpl implements AuthService {
+
+  private final UserService userService;
+  private final EmailService emailService;
+  private final UserRepository userRepository;
+  private final ModelMapper modelMapper;
+  private final BCryptPasswordEncoder bCryptPasswordEncoder;
+
+  @Override
+  public void sendVerificationEmail(String emailAddress) {
+    UserDto userDto = userService.findUserByEmail(emailAddress);
+
+    if (userDto.getVerified()) {
+      throw new UserException(UserExceptionType.USER_ALREADY_REGISTERED);
+    }
+
+    String token = JwtUtils.generateEmailVerificationToken(emailAddress);
+    Email email = EmailUtil.createVerificationEmail(userDto, token);
+
+    emailService.send(email);
+  }
+
+  @Override
+  public void sendPasswordResetEmail(String emailAddress) {
+    UserDto userDto = userService.findUserByEmail(emailAddress);
+
+    if (!userRepository.existsByEmail(emailAddress)) {
```

Comment:
Basicly if userDto is not null it exists right?

Reply:
Yeah, I checked it twice by accident, findUserByEmail already checks if the user exists, and throws an error if he doesn't.
Will remove the redundant check.